File: src/main/java/aurocosh/divinefavor/common/network/message/client/spirit_data/MessageSyncContracts.java
package aurocosh.divinefavor.common.network.message.client.spirit_data;

import aurocosh.divinefavor.DivineFavor;
import aurocosh.divinefavor.common.custom_data.player.PlayerData;
import aurocosh.divinefavor.common.custom_data.player.data.favor.SpiritData;
import aurocosh.divinefavor.common.network.base.WrappedClientMessage;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public class MessageSyncContracts extends WrappedClientMessage {
    private static String TAG_CONTRACTS = "Contracts";

    public NBTTagCompound tag;

    public MessageSyncContracts() {
    }

    public MessageSyncContracts(SpiritData data) {
        tag = getNbtTagCompound(data);
    }

    @Override
    @SideOnly(Side.CLIENT)
    protected void handleSafe() {
        EntityPlayer player = DivineFavor.proxy.getClientPlayer();
        SpiritData data = PlayerData.get(player).getSpiritData();
        setDataFromNBT(data, tag);
    }

    public static NBTTagCompound getNbtTagCompound(SpiritData instance) {
        final NBTTagCompound tag = new NBTTagCompound();
        tag.setTag(TAG_CONTRACTS, instance.serializeContracts());
        return tag;
    }

    public static void setDataFromNBT(SpiritData instance, NBTTagCompound nbt) {
        instance.deserializeContracts(nbt.getCompoundTag(TAG_CONTRACTS));
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/network/message/client/spirit_data/MessageSyncContracts.kt
package aurocosh.divinefavor.common.network.message.client.spirit_data

import aurocosh.divinefavor.DivineFavor
import aurocosh.divinefavor.common.custom_data.player.data.favor.SpiritData
import aurocosh.divinefavor.common.lib.extensions.divinePlayerData
import aurocosh.autonetworklib.network.base.WrappedClientMessage
import net.minecraft.nbt.NBTTagCompound
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

class MessageSyncContracts : WrappedClientMessage {
    var tag: NBTTagCompound = NBTTagCompound()

    constructor() {}

    constructor(data: SpiritData) {
        tag.setTag(TAG_CONTRACTS, data.serializeContracts())
    }

    @SideOnly(Side.CLIENT)
    override fun handleSafe() {
        val player = DivineFavor.proxy.clientPlayer
        player.divinePlayerData.spiritData.deserializeContracts(tag.getCompoundTag(TAG_CONTRACTS))
    }

    companion object {
        private const val TAG_CONTRACTS = "Contracts"
    }
}